app/views/documents/index.js.erb

```diff
@@ -0,0 +1,42 @@
+$("tbody").replaceWith("<tbody><%= j render partial: "documents/document", collection: @documents.select {|document|
+                    document.category == @category} %></tbody>");
+
+<%# I'd like to add the code between the comments to a different file, so that this file is organized better %>
```

Comment:
I thing you should be able to `import` any JS file here. But you might need to add it to the asset pipeline first.
Try to do that, but if it takes you more than 30mins, we'll do that together